src/screens/DiscoveryScreenV2/elements/DiscoveryTab.js

```diff
@@ -1,44 +1,64 @@
 
+import {useRoute} from '@react-navigation/core';
+import PropTypes from 'prop-types';
 import {colors} from '../../../utils/colors';
-import {fonts, normalizeFontSize} from '../../../utils/fonts';
+import {normalizeFontSize} from '../../../utils/fonts';
+import {setNavbarTitle} from '../../../context/actions/setMyProfileAction';
+import {Context} from '../../../context';
 
-const {width} = Dimensions.get('screen');
+const windowWidth = Dimensions.get('window').width;
+
+const DiscoveryTab = ({onChangeScreen, selectedScreen = 0, tabs}) => {
+  const [, dispatchNavbar] = React.useContext(Context).profile;
+  const route = useRoute();
 
-const DiscoveryTab = ({onChangeScreen, selectedScreen = 0}) => {
   const handleTabOnClicked = React.useCallback((index) => {
     Keyboard.dismiss();
     onChangeScreen(index);
   }, []);
 
-  const tabs = ['Users', 'Communities', 'Domains', 'News'];
+  const handleChangeTitle = (index = 0) => {
+    const title = ['Search Users', 'Your Communities', 'Your Domains'][index] || '';
+    setNavbarTitle(title, dispatchNavbar);
+  };
+
+  React.useEffect(() => {
+    if (route.name === 'Followings') handleChangeTitle();
+  }, []);
+
   return (
-    <>
-      <ScrollView
-        horizontal={true}
-        style={styles.tabContainer}
-        keyboardShouldPersistTaps="handled"
-        showsHorizontalScrollIndicator={false}
-        showsVerticalScrollIndicator={false}>
-        {tabs.map((item, index) => (
+    <ScrollView
+      horizontal={true}
+      style={styles.tabContainer}
+      keyboardShouldPersistTaps="handled"
+      showsHorizontalScrollIndicator={false}
+      showsVerticalScrollIndicator={false}>
+      {Object.keys(tabs).map((item, index) => {
+        if (route.name === 'Followings' && item === 'News') return null;
+        const totalItem = route.name === 'Followings' ? `(${tabs[item]})` : '';
+        return (
           <Pressable
             key={`tabItem-${item}`}
-            android_ripple={{
-              color: colors.gray1
-            }}
-            style={styles.tabItem(index)}
-            onPress={() => handleTabOnClicked(index)}>
+            android_ripple={{color: colors.gray1}}
+            style={[
+              styles.tabItem(route.name === 'Followings' ? 3 : 4),
+              index === selectedScreen ? styles.underlineFocus : {}
+            ]}
+            onPress={() => {
+              handleTabOnClicked(index);
+              handleChangeTitle(index);
+            }}>
             <View style={styles.tabItemContainer}>
               <Text style={index === selectedScreen ? styles.tabItemTextFocus : styles.tabItemText}>
-                {item}
+                {`${item} ${totalItem}`}
               </Text>
-              <View style={index === selectedScreen ? styles.underlineFocus : {}}></View>
             </View>
           </Pressable>
-        ))}
-      </ScrollView>
-    </>
+        );
+      })}
+    </ScrollView>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The rendering logic has been significantly changed. It now depends on the route name and the passed `tabs` prop. Make sure to test all possible combinations of props and routes to ensure the component behaves as expected. Also, it's not clear why `News` tab is excluded when route name is `Followings`. If this is a business requirement, consider adding a comment to explain it.
```diff
- const tabs = ['Users', 'Communities', 'Domains', 'News'];
+ {Object.keys(tabs).map((item, index) => {
+   if (route.name === 'Followings' && item === 'News') return null;
+   const totalItem = route.name === 'Followings' ? `(${tabs[item]})` : '';
+   return (
+     <Pressable
+       key={`tabItem-${item}`}
+       android_ripple={{color: colors.gray1}}
+       style={[
+         styles.tabItem(route.name === 'Followings' ? 3 : 4),
+         index === selectedScreen ? styles.underlineFocus : {}
+       ]}
+       onPress={() => {
+         handleTabOnClicked(index);
+         handleChangeTitle(index);
+       }}>
+       <View style={styles.tabItemContainer}>
+         <Text style={index === selectedScreen ? styles.tabItemTextFocus : styles.tabItemText}>
+           {`${item} ${totalItem}`}
+         </Text>
+       </View>
+     </Pressable>
+   );
+ })}
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->